tests/apps/test_similarity.py

```diff
@@ -222,142 +223,224 @@ def test_event_cardinality(photons, max_count, modes, expected):
-        is equal to 1/5, i.e., one over the number of samples in the orbit [1,1,1,1] for 5 modes."""
+        is equal to 1/5, i.e., one over the cardinality of the orbit [1,1,1,1] for 5 modes."""
         graph = nx.complete_graph(5)
         with monkeypatch.context() as m:
             m.setattr(
-                "strawberryfields.backends.gaussianbackend.GaussianState.fock_prob",
+                "strawberryfields.backends.gaussianbackend.states.GaussianState.fock_prob",
                 lambda *args, **kwargs: 0.2,
             )
-
             assert np.allclose(similarity.prob_orbit_mc(graph, [1, 1, 1, 1]), 1.0)
 
     def test_prob_vacuum_orbit(self):
         """Tests if the function gives the right probability for the empty orbit when the GBS
         device has been configured to have zero mean photon number."""
-        graph = nx.complete_graph(10)
-
-        assert similarity.prob_orbit_mc(graph, [], 0) == 1.0
-
-    def test_loss(self, monkeypatch):
-        """Test if function correctly creates the SF program for lossy GBS."""
         graph = nx.complete_graph(5)
-        mock_eng_run = mock.MagicMock()
-
-        with monkeypatch.context() as m:
-            m.setattr(sf.LocalEngine, "run", mock_eng_run)
-            similarity.prob_orbit_mc(graph, [1, 1, 1, 1], samples=1, loss=0.5)
-            p_func = mock_eng_run.call_args[0][0]
-
-        assert isinstance(p_func.circuit[1].op, sf.ops.LossChannel)
-
-    def test_no_loss(self, monkeypatch):
-        """Test if function correctly creates the SF program for GBS without loss."""
-        graph = nx.complete_graph(5)
-        mock_eng_run = mock.MagicMock()
-
-        with monkeypatch.context() as m:
-            m.setattr(sf.LocalEngine, "run", mock_eng_run)
-            similarity.prob_orbit_mc(graph, [1, 1, 1, 1], samples=1)
-            p_func = mock_eng_run.call_args[0][0]
-
-        assert not all([isinstance(op, sf.ops.LossChannel) for op in p_func.circuit])
+        assert similarity.prob_orbit_mc(graph, [], 0) == 1.0
 
-    def test_all_loss(self, monkeypatch):
+    def test_max_loss(self):
         """Test if function samples from the vacuum when maximum loss is applied."""
-        dim = 5
-        graph = nx.complete_graph(dim)
-        mock_eng_run = mock.MagicMock()
-
-        with monkeypatch.context() as m:
-            m.setattr(sf.LocalEngine, "run", mock_eng_run)
-            similarity.prob_orbit_mc(graph, [1, 1, 1, 1], samples=1, loss=1)
-            p_func = mock_eng_run.call_args[0][0]
-
-        eng = sf.LocalEngine(backend="gaussian")
-
-        state = eng.run(p_func).state
-        cov = state.cov()
-        disp = state.displacement()
-
-        assert np.allclose(cov, 0.5 * state.hbar * np.eye(2 * dim))
-        assert np.allclose(disp, np.zeros(dim))
+        graph = nx.complete_graph(5)
+        assert similarity.prob_orbit_mc(graph, [1, 1, 1, 1], samples=1, loss=1) == 0.0
 
 
 class TestProbEventMC:
```

Comment:
Also needs to test that correct values are calculated for non-trivial specific cases